proseco/guide.py

```diff
@@ -0,0 +1,551 @@
+
+
+def check_column_spoilers(cand_stars, ok, stars, n_sigma):
+    """
+    For each candidate, check for stars 'MagDiff' brighter and within 'Separation' columns
+    between the star and the readout register, i.e. Column Spoilers.
+
+    :param cand_stars: Table of candidate stars
+    :param ok: mask on cand_stars describing still "ok" candidates
+    :param stars: Table of AGASC stars
+    :param n_sigma: multiplier used when checking mag with mag_err
+    :returns: bool mask on cand_stars marking column spoiled stars
+    """
+    column_spoiled = np.zeros_like(ok)
+    for idx, cand in enumerate(cand_stars):
+        if not ok[idx]:
+            continue
+        dcol = cand['col'] - stars['col'][~stars['offchip']]
+        direction = stars['row'][~stars['offchip']] / cand['row']
+        pos_spoil = ((np.abs(dcol) <= (GUIDE_CHAR.col_spoiler['Separation']))
+                     & (direction > 1.0))
+        if not np.count_nonzero(pos_spoil) >= 1:
+            continue
+        mag_errs =  (n_sigma * np.sqrt(cand['mag_err'] ** 2
+                                       + stars['mag_err'][~stars['offchip']][pos_spoil] ** 2))
+        dm = (cand['MAG_ACA'] - stars['MAG_ACA'][~stars['offchip']][pos_spoil] + mag_errs)
+        if np.any(dm > GUIDE_CHAR.col_spoiler['MagDiff']):
+            column_spoiled[idx] = True
+    return column_spoiled
+
+
+def get_imposter_mags(cand_stars, dark, dither):
+    """
+    Get "pseudo-mag" of max pixel value in each candidate star region
+
+    :param cand_stars: Table of candidate stars
+    :param dark: full CCD dark map
+    :param dither: observation dither to be used to determine pixels a star could use
+    :returns: array of magnitudes of length cand_stars
+    """
+    def get_ax_range(r, extent):
+        # Should come back to this and do something smarter
+        # but right now I just want things that bin nicely 2x2
+        rminus = int(np.floor(r - row_extent))
+        rplus = int(np.ceil(r + row_extent))
+        if (np.floor(r) != np.ceil(r)):
+            if r - np.floor(r) > .5:
+                rplus += 1
+            else:
+                rminus -= 1
+        return rminus, rplus
+
+    pixmags = []
+    # Define the 1/2 pixel region as half the 8x8 plus dither
+    row_extent = np.ceil(4 + dither[0] * GUIDE_CHAR.ARC_2_PIX)
+    col_extent = np.ceil(4 + dither[1] * GUIDE_CHAR.ARC_2_PIX)
+    for idx, cand in enumerate(cand_stars):
+        rminus, rplus = get_ax_range(cand['row'], row_extent)
+        cminus, cplus = get_ax_range(cand['col'], col_extent)
+        pix = np.array(dark.aca[rminus:rplus, cminus:cplus])
+        pixmax = max(np.max(bin2x2(pix)),
+                     np.max(bin2x2(pix[1:-1])),
+                     np.max(bin2x2(pix[:,1:-1])),
+                     np.max(bin2x2(pix[1:-1,1:-1])))
+        pixmax_mag = count_rate_to_mag(pixmax)
+        pixmags.append(pixmax_mag)
+    return np.array(pixmags)
+
+
+def get_pixmag_for_offset(cand_mag, offset):
```

Comment:
Instead of setting the limit for a star's "spoiler pixel" to be a magnitude (if the star has a pixel in its region over a pseudo magnitude threshold it would be excluded), I worked with the idea that the limit is a centroid offset and then did the ~worst case calculation that should give the magnitude *for* that offset.  I don't think this is crazy, but it is different from the other limits in the characteristics.